### PR TITLE
fix(redpanda): wait for

### PR DESCRIPTION
**Pull request by stevenh**
Wait for the admin interface to response to HTTP to avoid failures in configuring the instance when its not fully ready.

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

File: modules/redpanda/redpanda.go [hunk 1]
```diff
@@ -192,34 +192,53 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
...
+	err = ctr.CopyToContainer(ctx, nodeConfig, filepath.Join(redpandaDir, "redpanda.yaml"), 0o600)
 	if err != nil {
-		return c, fmt.Errorf("failed to copy redpanda.yaml into container: %w", err)
+		return c, fmt.Errorf("copy to container: %w", err)
 	}
 
 	// 8. Wait until Redpanda is ready to serve requests.
+	waitHTTP := wait.ForHTTP(defaultAdminAPIPort).
+		WithStatusCodeMatcher(func(status int) bool {
+			return status == http.StatusNotFound
```
Comment (mdelapenya) [inline modules/redpanda/redpanda.go:217]: Is it status not found?

Comment (stevenh) [reply modules/redpanda/redpanda.go:217]: We're making a request to the root which doesn't have a handler, so yer this looks weird but it is right.

Comment (mdelapenya) [reply modules/redpanda/redpanda.go:217]: Can we add a comment about that? 🙏

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Pull request merged**